cmd/argocd-server/commands/argocd_server.go

```diff
@@ -164,8 +166,6 @@ func NewCommand() *cobra.Command {
 			stats.RegisterHeapDumper("memprofile")
 
 			for {
-				ctx := context.Background()
-				ctx, cancel := context.WithCancel(ctx)
```

Comment:
This can't be moved outside this loop.

Reply:
If this cannot be moved, then the `InitTracer` call should be inside the loop as well? Because it needs the server context

Reply:
Correct. A new context needs to be built on every loop iteration and the `InitTracer` call needs to happen somewhere inside this loop.

Reply:
This seems a little bit weird to me. But anyway I updated the code.